src/views/pages/dashboard/invoice/DashboardInvoicePageTs.ts

```diff
@@ -70,7 +47,9 @@ export interface BalanceEntry {
   subscriptions() {
     const qrCode$ = this
       .$watchAsObservable('transactionQR', {immediate: true})
-      .pipe(pluck('newValue'),
+      .pipe(
+        debounceTime(500),
```

Comment:
try throttleTime

Reply:
The difference between `throttleTime` and `debounceTime` :

When the value is 500(ms).

- throttleTime

It will call this function once every 500ms, no matter the value changed or not.

- debounceTime

When the value of `transactionQR` changed, it will call this function after 500ms, no matter the value changed how many times in next 500ms.

When user inputing the form, the transaction will change with user inputing. So the value of `transactionQR` will change a lot times, but we just need the last transaction to create the QR. For saving the times of computing, I think `debounceTime` is better.

I tried `throttleTime`, I feel not good.

Reply:
did you put the leading and trailing options? 
 throttleTime(EVENTS_THROTTLING_TIME, asyncScheduler, {leading: true, trailing: true}),

Reply:
No, I just `throttleTime(500)`.

Reply:
ok, please try throttleTime(500, asyncScheduler, {leading: true, trailing: true})

import {asyncScheduler} from 'rxjs'

throttle time will not call this function every 500 ms

with this options, it calls the function as soon as the form will be changed (debounceTime will wait for 500ms), and the next call will be the one triggered by the last change that occurred from the first call to 500ms after the first call, which is why it is better in terms of UX

https://rxmarbles.com/#debounceTime
https://rxmarbles.com/#throttleTime

But tbh, I think we could just not throttle nor debounce

Reply:
debouncing and throttling should be done for expensive operations, this one is not expensive

Reply:
Finally, I removed it.
But we must be attention to the users whoes computer is not good. Maybe creating QR so frequently is not good.